packages/utils/test/validators.spec.ts

```diff
@@ -328,3 +328,914 @@ describe("Validators", () => {
+          chains,
+          methods,
+          events,
+          accounts,
+        },
+      },
+    });
+
+    const expected = {
+      eip155: {
+        chains: ["eip155:1", "eip155:2"],
+        methods: ["personal_sign", "eth_sendTransaction", "eth_signTransaction"],
+        events,
+        accounts: [
+          "eip155:1:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+          "eip155:2:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+        ],
+      },
+    };
+    expect(approvedNamespaces).to.deep.eq(expected);
+  });
+  it("should build namespaces (config 11 - multiple namespaces - required)", () => {
+    const required = {
+      "eip155:1": {
+        events: ["chainChanged"],
+        methods: ["personal_sign", "eth_sendTransaction"],
+      },
+      cosmos: {
+        chains: ["cosmos:cosmoshub-4"],
+        events: ["cosmos_event"],
+        methods: ["cosmos_method"],
+      },
+    };
+    const optional = {
+      eip155: {
+        chains: ["eip155:1", "eip155:2"],
+        events: ["chainChanged", "accountsChanged"],
+        methods: [
+          "personal_sign",
+          "eth_sendTransaction",
+          "eth_signTransaction",
+          "eth_signTypedData",
+        ],
+      },
+    };
+
+    const chainsEip = ["eip155:1", "eip155:2", "eip155:4"];
+    const chainsCosmos = ["cosmos:cosmoshub-4"];
+    const methods = ["personal_sign", "eth_sendTransaction", "eth_signTransaction"];
+    const events = ["chainChanged", "accountsChanged"];
+    const accountsEip = [
+      "eip155:1:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+      "eip155:2:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+      "eip155:4:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+    ];
+    const accountsCosmos = ["cosmos:cosmoshub-4:cosmos1hsk6jryyqjfhp5dhc55tc9jtckygx0eph6dd02"];
+    const eventsCosmos = ["cosmos_event"];
+    const methodsCosmos = ["cosmos_method"];
+
+    const approvedNamespaces = buildApprovedNamespaces({
+      requiredNamespaces: required,
+      optionalNamespaces: optional,
+      supportedNamespaces: {
+        eip155: {
+          chains: chainsEip,
+          methods,
+          events,
+          accounts: accountsEip,
+        },
+        cosmos: {
+          chains: chainsCosmos,
+          methods: methodsCosmos,
+          events: eventsCosmos,
+          accounts: accountsCosmos,
+        },
+      },
+    });
+
+    const expected = {
+      eip155: {
+        chains: ["eip155:1", "eip155:2"],
+        methods: ["personal_sign", "eth_sendTransaction", "eth_signTransaction"],
+        events,
+        accounts: [
+          "eip155:1:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+          "eip155:2:0x57f48fAFeC1d76B27e3f29b8d277b6218CDE6092",
+        ],
+      },
+      cosmos: {
+        chains: chainsCosmos,
+        methods: methodsCosmos,
+        events: eventsCosmos,
+        accounts: accountsCosmos,
+      },
+    };
+    expect(approvedNamespaces).to.deep.eq(expected);
+  });
+  it.fails(
```

Comment:
TIL about there being an explicit `fails` method in vitest.

If we’re expecting specific error returns is there a way to check that to avoid false positives on these? Eg these throw but they actually all throw for the same reason due to a bug etc

Reply:
Actually maybe overkill since the only cause of throwing is `isConformingNamespaces` which has its own tests.